daemon/main.cpp

```diff
@@ -642,20 +640,30 @@ void handle_pid_daemon(std::string const &pidf, bool daemon)
 
     if (daemon)
     {
-        if (pipe(waitFdPair) < 0)
+        pid = fork();
+        if (pid && pid < 0)
         {
-            LogError << "daemonize: pipe() failed.";
+            LogError << "daemonize: fork() failed.";
             exit(1);
         }
-        pid = fork();
-        if (pid)
+
+        if (pid > 0)
         {
-            if (pid < 0)
-            {
-                LogError << "daemonize: fork() failed.";
-                exit(1);
-            }
+            // Kill the parent process
+            exit(0);
         }
+
+        pid_t sid;
+        sid = setsid();
+        if (sid > 0)
+        {
+            LogError << "daemonize: setsid() failed.";
+            exit(1);
+        }
+
+        ::close(0);
```

Comment:
The old code explicitly set these to go to /dev/null, which avoids EBADF when the process actually writes. Don't know if this really matters, but it's a cleanliness thing.